### DISCUSSION TITLE
How to calculate the font size of srt subtitle in mpv

### DISCUSSION THREAD

**Discussion by ahaoboy** [mpv-player/mpv#15033] [Q&A]
I am developing a translation plugin for mpv and want to be able to display two languages ​​at the same time, with the primary language using a normal font size and the secondary language using a smaller font size.

The default sub-font-size of mpv is 55, but it is too big

```
1
00:00:05,171 --> 00:00:06,756
<font size="55px">这对你来说应该很容易，对吧，瑞克？</font>
<font size="27px">It should be easy for you, right, Rick?</font>
```
![image](https://github.com/user-attachments/assets/f2f1a521-03db-40e6-a1fc-51f45aac191f)

![image](https://github.com/user-attachments/assets/e14c13a7-a8d1-4552-9cfa-2dd7d3f2d9d0)

I cut it in half and it seems to be about the same size, but why do I need to cut it in half? Is this ratio different in other devices? I want to know if there is any way to convert the sub-font-size in mpv to the font size in srt
![image](https://github.com/user-attachments/assets/e30e1818-c47d-45c8-a38d-cefc753e6f6b)

**Comment by mrfragger** [comment]
I believe default is sub-scale=2.5
if you put sub-scale=1 
then you can do sub-font-size= to have a much wider range

**Reply by ahaoboy** [reply]
> I believe default is sub-scale=2.5 if you put sub-scale=1 then you can do sub-font-size= to have a much wider range

However, sub-scale is 1. I wonder where the 2.5 is calculated from? Is it a constant? The window size is 1920x1080

**Comment by guidocella** [comment]
`--sub-scale-by-window=no`

**Reply by ahaoboy** [reply]
> `--sub-scale-by-window=no`

This controls whether the font scales with the window size, which does not seem to solve this problem
